src/js/components/LayerPanel/TerraIControls.js

```diff
@@ -84,10 +78,20 @@ export default class TerraIControls extends Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    //ensure the startDate is an empty object and not a Date Object
+    if ((prevProps.startDate !== this.props.startDate && this.props.startDate.constructor === Object)
+      && prevProps.endDate !== this.props.endDate && this.props.endDate.constructor === Object) {
+      this.fromPicker.set('select', this.state.min);
+      this.toPicker.set('select', this.state.max);
+    }
+  }
```

Comment:
Terra I works a little different than GLAD here. Since the min and max dates are set when the layer is initialized (instead of in the initial state), we can't just reset to the initial state when the clear button is clicked. We need a way to tell the component that the clear button has been pressed, so I chose to pass empty objects for `startDate` and `endDate`. Then we need to check if the component received an empty object, and update the calendar (and layer def) accordingly.